--- shadow-4.0.0~/libmisc/chkname.c	Thu Apr 16 23:57:43 1998
+++ shadow-4.0.0/libmisc/chkname.c	Mon Mar 18 19:54:33 2002
@@ -22,19 +22,17 @@
 good_name(const char *name)
 {
 	/*
-	 * User/group names must start with a letter, and may not
-	 * contain colons, commas, newlines (used in passwd/group
-	 * files...) or any non-printable characters.
+	 * User/group names must match [a-z_][a-z0-9_-]*
 	 */
-	if (!*name || !isalpha(*name))
+	if (!*name || !((*name >= 'a' && *name <= 'z') || *name == '_'))
 		return 0;
 
-	while (*name) {
-		if (*name == ':' || *name == ',' ||
-		    *name == '\n' || !isprint(*name))
+	while (*++name) {
+		if (!((*name >= 'a' && *name <= 'z') ||
+		    (*name >= '0' && *name <= '9') ||
+		    *name == '_' || *name == '-' ||
+		    (*name == '$' && name[1] == '\0')))
 			return 0;
-
-		name++;
 	}
 
 	return 1;
@@ -53,7 +51,7 @@
 	 * User names are limited by whatever utmp can
 	 * handle (usually max 8 characters).
 	 */
-	if (strlen(name) > sizeof(ut.ut_user))
+	if (strlen(name) >= sizeof(ut.ut_user))
 		return 0;
 
 	return good_name(name);
