--- shadow-4.0.4.1/configure.in.orig	2004-11-10 19:35:57 +0300
+++ shadow-4.0.4.1/configure.in	2004-11-10 19:35:57 +0300
@@ -175,14 +175,7 @@
 	fi
 done
 
-AC_MSG_CHECKING(location of the passwd program)
-if test -f /usr/bin/passwd; then
-	passwd_dir=/usr/bin
-else
-	passwd_dir=/bin
-fi
-AC_DEFINE_UNQUOTED(PASSWD_PROGRAM, "$passwd_dir/passwd")
-AC_MSG_RESULT($passwd_dir)
+AC_DEFINE_UNQUOTED(PASSWD_PROGRAM, "/usr/bin/passwd")
 
 dnl XXX - quick hack, should disappear before anyone notices :).
 AC_DEFINE(SHADOWPWD)
