--- shadow-4.0.0/Makefile.am.orig	2004-03-25 21:35:05 +0300
+++ shadow-4.0.0/Makefile.am	2004-03-25 21:55:57 +0300
@@ -4,5 +4,5 @@
 
 AUTOMAKE_OPTIONS = 1.4 foreign
 
-SUBDIRS = intl po man libmisc lib src \
+SUBDIRS = po man libmisc lib src \
 	contrib debian doc etc
--- shadow-4.0.0/configure.in.orig	2004-03-25 21:35:05 +0300
+++ shadow-4.0.0/configure.in	2004-03-25 22:04:58 +0300
@@ -32,7 +32,7 @@
 	utime.h ulimit.h sys/resource.h gshadow.h shadow.h lastlog.h \
 	rpc/key_prot.h)
 
-AC_CHECK_FUNCS(a64l fchmod fchown fsync getgroups gethostname getspnam \
+AC_CHECK_FUNCS(a64l fchmod fchown fsync getgroups gethostname getspnam setlocale \
 	gettimeofday getusershell getutent initgroups lchown lckpwdf lstat \
 	memcpy memset mkstemp setgroups sigaction strchr updwtmp updwtmpx)
 
@@ -277,11 +277,11 @@
 fi
 
 ALL_LINGUAS="cs el fr pl sv uk"
-AM_GNU_GETTEXT
+AM_GNU_GETTEXT([external])
+AM_GNU_GETTEXT_VERSION(0.10.38)
 
 AC_OUTPUT([
 	Makefile
-	intl/Makefile
 	po/Makefile.in
 	doc/Makefile
 	man/Makefile
