--- shadow-4.0.4.1/src/useradd.c.orig	2004-11-10 19:30:27 +0300
+++ shadow-4.0.4.1/src/useradd.c	2004-11-10 19:30:27 +0300
@@ -1956,7 +1956,8 @@ int main (int argc, char **argv)
 	if (mflg) {
 		create_home ();
 		create_mailbox ();
-		copy_tree (def_template, user_home, user_id, user_gid, 1);
+		copy_tree ((access(def_template,R_OK|X_OK)?SKEL_DIR:def_template),
+		user_home, user_id, user_gid, 1);
 	}
 
 	nscd_flush_cache ("passwd");
