--- shadow-4.0.0/libmisc/copydir.c.orig	2002-08-14 19:15:05 +0400
+++ shadow-4.0.0/libmisc/copydir.c	2002-08-14 19:29:51 +0400
@@ -204,7 +204,7 @@
 			 * the user and then recursively copy that directory.
 			 */
 
-			mkdir (dst_name, sb.st_mode & 0777);
+			mkdir (dst_name, sb.st_mode & 0700);
 			chown (dst_name, uid == (uid_t) -1 ? sb.st_uid:uid,
 				gid == (gid_t) -1 ? sb.st_gid:gid);
 
@@ -276,10 +276,10 @@
 		 */
 
 		if (!S_ISREG(sb.st_mode)) {
-			if (mknod (dst_name, sb.st_mode & ~07777, sb.st_rdev) ||
+			if (mknod (dst_name, sb.st_mode & ~07700, sb.st_rdev) ||
 				chown (dst_name, uid == (uid_t) -1 ? sb.st_uid:uid,
 					gid == (gid_t) -1 ? sb.st_gid:gid) ||
-					chmod (dst_name, sb.st_mode & 07777)) {
+					chmod (dst_name, sb.st_mode & 07700)) {
 				err++;
 				break;
 			}
@@ -301,7 +301,7 @@
 		if ((ofd = open (dst_name, open_flags, 0)) < 0 ||
 			chown (dst_name, uid == (uid_t) -1 ? sb.st_uid:uid,
 					gid == (gid_t) -1 ? sb.st_gid:gid) ||
-				chmod (dst_name, sb.st_mode & 07777)) {
+				chmod (dst_name, sb.st_mode & 07700)) {
 			if (preserve && (errno == EEXIST)) {
 				close (ifd);
 				continue;
